src/python/pants/bin/exception_sink.py

```diff
@@ -0,0 +1,43 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir
+from pants.util.memo import memoized_classproperty
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  @memoized_classproperty
+  def instance(cls):
+    return cls()
+
+  def __init__(self):
+    self._cached_destination = None
```

Comment:
This should probably default to `cwd` so that it is always useable, even before the destination has been set the first time.

Also I'm not sure it's really a "cached" destination... maybe just "destination".

Reply:
That sounds great, implementing this and the below now.